src/vs/editor/browser/widget/multiDiffEditorWidget/multiDiffEditorViewModel.ts

```diff
@@ -27,6 +27,12 @@ export class MultiDiffEditorViewModel extends Disposable {
 		}
 	}
 
+	public collapseAll(): void {
+		for (const d of this.items.get()) {
+			d.collapsed.set(true, undefined);
```

Comment:
This should run in a transaction (rather than passing in `undefined`), as otherwise every `collapsed.set` will cause a re-layout.